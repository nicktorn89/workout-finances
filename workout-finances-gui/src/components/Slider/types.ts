export interface SliderProps {
  date?: Date;
}
