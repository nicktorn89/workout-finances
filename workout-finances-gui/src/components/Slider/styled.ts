import styled from 'styled-components';
import { Typography } from '@material-ui/core';

// @ts-nocheck

export const SliderContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 65px;
`;

export const DateTitle: any = styled(Typography)`
  margin: 0 2rem !important;
  font-size: 18px !important;
  font-weight: 700 !important;
`;
