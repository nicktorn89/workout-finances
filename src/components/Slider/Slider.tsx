import IconButton from '@material-ui/core/IconButton';
import React, { memo } from 'react';
import { SliderProps } from './types';
import { SliderContainer, DateTitle } from './styled';
import ArrowLeftRounded  from '@material-ui/icons/ArrowLeftRounded';
import ArrowRightRounded  from '@material-ui/icons/ArrowRightRounded';

const Slider: React.FC<SliderProps> = ({}) => {
  return(
    <SliderContainer>
      <IconButton>
        <ArrowLeftRounded />
      </IconButton>

      <DateTitle>
        1 половина марта
      </DateTitle>

      <IconButton>
        <ArrowRightRounded />
      </IconButton>
    </SliderContainer>
  );
};

export default memo(Slider);
