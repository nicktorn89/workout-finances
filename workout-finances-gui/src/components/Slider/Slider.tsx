import IconButton from '@material-ui/core/IconButton';
import React, { memo } from 'react';
import { SliderProps } from './types';
import { SliderContainer, DateTitle } from './styled';
import ArrowLeftRounded  from '@material-ui/icons/ArrowLeftRounded';
import ArrowRightRounded  from '@material-ui/icons/ArrowRightRounded';

const Slider: React.FC<SliderProps> = ({ date }) => {
  const month = date && date.getMonth();

  return(
    <SliderContainer>
      <IconButton>
        <ArrowLeftRounded />
      </IconButton>

      <DateTitle>
        {date ? `1 половина ${month}` : '1 половина'}
      </DateTitle>

      <IconButton>
        <ArrowRightRounded />
      </IconButton>
    </SliderContainer>
  );
};

export default memo(Slider);
