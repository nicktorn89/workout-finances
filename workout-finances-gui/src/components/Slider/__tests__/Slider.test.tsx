import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Slider from '../Slider';

describe('<Slider />', () => {
  const SliderComponent = shallow(<Slider />);
  const SliderComponentToJson = shallowToJson(SliderComponent);
  
  it('Should be equal to snapshot', () => {
    expect(SliderComponentToJson).toMatchSnapshot();
  });
});
