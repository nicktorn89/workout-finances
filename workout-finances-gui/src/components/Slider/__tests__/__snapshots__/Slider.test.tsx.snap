// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Should be equal to snapshot 1`] = `
<styled.div>
  <WithStyles(IconButton)>
    <pure(ArrowLeftRoundedIcon) />
  </WithStyles(IconButton)>
  <Styled(WithStyles(Typography))>
    1 половина
  </Styled(WithStyles(Typography))>
  <WithStyles(IconButton)>
    <pure(ArrowRightRoundedIcon) />
  </WithStyles(IconButton)>
</styled.div>
`;
